packages/material-ui/src/Slider/Slider.js

```diff
@@ -714,6 +723,10 @@ const Slider = React.forwardRef(function Slider(props, ref) {
   }, [disabled, handleTouchEnd, handleTouchMove]);
 
   const handleMouseDown = useEventCallback((event) => {
+    // don't listen to right clicks
+    if (event.buttons === 2) {
```

Comment:
Why not?

```suggestion
    if (event.buttons !== 1) {
```

https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons

Reply:
Note  that `buttons` is a bitmap meaning that primary and secondary button press would result in `event.buttons === 3`. 

If you only want to ignore right clicks 
```suggestion
    if (event.button === 1) {
```
would match that behavior. Though we probably just want to consider clicks with the primary button and ignore anything else so 
```suggestion
    if (event.button !== 0) {
```
would be the way to go.

Reading `buttons` should only ever happen using bit-operators. For the rest `event.button` is more descriptive.

Reply:
> Why not?
> 
> https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons

because unless you prevent the default behavior it opens a right click menu there's no scenerio where you'd want that. perhaps we'd want to make an option to allow it and also prevent the context menu from opening so that people can do _things_ things it. i can't personally think of a usecase where you'd want the slider to activate on right click.

Reply:
> Note that `buttons` is a bitmap meaning that primary and secondary button press would result in `event.buttons === 3`.
> 
> If you only want to ignore right clicks
> 
> would match that behavior. Though we probably just want to consider clicks with the primary button and ignore anything else so
> 
> would be the way to go.
> 
> Reading `buttons` should only ever happen using bit-operators. For the rest `event.button` is more descriptive.

Wouldn't you want left + right to activate it though? I only figured it'd make sense to explicitly hard block when it's explicitly only a right click. What's the reasoning to also block combinations of right? It seems reasonable overall to me, but a bit more heavy handed, and it's already sort of an opinionated change..

Reply:
perhaps on mousedown it should be `.button` and for mousemove it should be `buttons` 🤔

Reply:
> Wouldn't you want left + right to activate it though?

mouse 4 and 5 are usually for page navigation. It'd be odd if you prevent that because you just happened to hover over the slider. Native sliders also ignore mouse 4 or 5. Currently we start dragging but upon release button 4 navigation still starts. 

But more importantly: comment and code don't match. This is problematic for maintaining this code.

Reply:
> It'd be odd if you prevent that because you just happened to hover over the slider. 

I have tested it, it prevents navigation if you click on the previous button of the mouse while hovering the slider 🙈.